src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import classNames from 'classnames';
+
+import { Person } from '../../types/Person';
+
+type Props = {
+  options: Person[],
+  onSelected: (person: Person) => void,
+  delay: number,
+};
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function debounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+export const Dropdown: React.FC<Props> = ({
+  options,
+  onSelected,
+  delay,
+}) => {
+  const [isListShowed, setIsListShowed] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
why do we still need this function if we can just call debounce in handleInputChange